### ISSUE TITLE
clickCallback of svelvet nodes is called even when clicking outside of the Svelvet Component

### ISSUE THREAD

**Issue by msze400** [open-source-labs/Svelvet#158]
I'm currently running into an issue where after clicking on a node, the clickCallBack associated with the node seems to persist. After the first click of the node, when the user clicks on anywhere outside the svelvet component window the clickCallBack seems to continue to fire again. It was recently working find similar to the demo pieces listed on the example page but within the last 10-15 days this error seemed to pop up. 

Has there been any new releases that might coincide with this problem?

**Comment by msze400** [comment]
I think I found what was causing our issue. In the file that describes the Node event handlers, [this condition](https://github.com/open-source-labs/Svelvet/blob/main/NPM%20Package/svelvet/Nodes/index.svelte#L43) determines when the clickCallback of a node is called. From what I could tell, the only time that `$nodeIdSelected` gets modified is by the "mousedown" event. This ultimately was causing the condition to evaluate to true on every click after clicking on a node in the graph, even when the next click was not on a node.

 

As a workaround, I set $nodeIdSelected equal to null after onNodeClick executes (on line 44-45) and that seems to resolve my issue. Has anyone else experienced this problem or can see a more suitable fix for this issue?
